src/shared/math.ts

```diff
@@ -0,0 +1,135 @@
+/** md
+## Math utilities for numberscope
+
+The primary resource for doing math inside the frontscope code is the
+[mathjs](http://mathjs.org) package, which this repository depends on.
+The `shared/math` module of Numberscope provides some additional utilities
+aimed at making common mathematical operations more convenient, especially
+when working with bigints.
+
+Note that currently every one of these functions accepts either
+`number` or `bigint` inputs for all arguments and simply converts them
+to bigints.
+
+### Example usage
+```ts
+import {
+    safeNumber,
+    floorSqrt,
+    modulo,
+    powmod,
+    natlog
+} from '../shared/math'
+try {
+  const myNumber = safeNumber(9007199254740992n)
+} catch (err: unknown) {
+  console.log('This will always be logged')
+}
+try {
+  const myNumber = safeNumber(9007n)
+} catch (err: unknown) {
+  console.log('This never will be and myNumber will be 9007')
+}
```

Comment:
This doesn't seem grammatical?  Will be what? Logged?

Reply:
Correct, but I will insert the word.

Reply:
Done.